sections/exchange/hooks/useExchange.tsx

```diff
@@ -718,6 +727,46 @@ const useExchange = ({
 		}
 	};
 
+	const handleSettle = async () => {
+		if (synthetixjs != null && gasPrice != null) {
+			setTxError(null);
+			setTxSettleModalOpen(true);
+
+			try {
+				const gasPriceWei = gasPriceInWei(gasPrice);
+				const gasInfo = await getGasEstimateForExchange(gasPriceWei);
+				setGasInfo(gasInfo);
+				const gas = {
+					gasPrice: gasPriceWei,
+					gasLimit: gasInfo?.limit,
+				};
+
+				// send transaction
+				const tx = await synthetixjs.contracts.Exchanger.settle(
+					walletAddress,
+					ethers.utils.formatBytes32String(baseCurrencyKey as string),
+					gas
+				);
+
+				if (tx != null) {
+					monitorTransaction({
+						txHash: tx.hash,
+						onTxConfirmed: () => {
```

Comment:
> The current UI is unable to refresh automatically after the settlement. Instead, it requires manual refresh (e.g., click the arrows in the middle). So I need some help to make it better.
Do we want to show the "submitting order" after clicking the "settle" button? The button text won't change right now.

In `onTxConfirmed` you could also try to invalidate the `useNumEntriesQuery` query. This should hopefully return 0 now that all transactions have been settled.

https://react-query.tanstack.com/guides/query-invalidation

Reply:
@JChiaramonte7 Thanks for your input. It reminds me to use `refetch()` to update the state before and after the settlement. The new change has been pushed.